http/client/src/main/java/org/eclipse/rdf4j/http/client/SharedHttpClientSessionManager.java

```diff
@@ -120,15 +128,45 @@ private CloseableHttpClient createHttpClient() {
 
 	@Override
 	public SPARQLProtocolSession createSPARQLProtocolSession(String queryEndpointUrl, String updateEndpointUrl) {
-		SPARQLProtocolSession session = new SPARQLProtocolSession(getHttpClient(), executor);
+		SPARQLProtocolSession session = new SPARQLProtocolSession(getHttpClient(), executor) {
+
+			{
```

Comment:
This is unusual syntax. Is there anything preventing it being done in a more usual way?